From 7b06c5415f24fe7d20253b101a6462d1944854d0 Mon Sep 17 00:00:00 2001
From: Lukas Slebodnik <lslebodn@redhat.com>
Date: Mon, 11 Jan 2016 19:13:06 +0100
Subject: [PATCH 71/86] UTIL: Use sss_vdebug_fn for callbacks
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Reviewed-by: Pavel Březina <pbrezina@redhat.com>
(cherry picked from commit f6c1f6a561bdd5b4bba03c02988a724da3dad387)
(cherry picked from commit 86ba562d09e1ee9aa55819d748722c2d3ac00600)
---
 src/util/debug.c        | 13 ++-----------
 src/util/sss_semanage.c | 14 +++-----------
 2 files changed, 5 insertions(+), 22 deletions(-)

diff --git a/src/util/debug.c b/src/util/debug.c
index 1b7aef467ff336e3316ab39f3a7d0b22ff3a0ed0..03dc26d6d4e976866a3f9395598840e28997f69a 100644
--- a/src/util/debug.c
+++ b/src/util/debug.c
@@ -285,8 +285,6 @@ void ldb_debug_messages(void *context, enum ldb_debug_level level,
                         const char *fmt, va_list ap)
 {
     int loglevel = SSSDBG_UNRESOLVED;
-    int ret;
-    char * message = NULL;
 
     switch(level) {
     case LDB_DEBUG_FATAL:
@@ -303,16 +301,9 @@ void ldb_debug_messages(void *context, enum ldb_debug_level level,
         break;
     }
 
-    ret = vasprintf(&message, fmt, ap);
-    if (ret < 0) {
-        /* ENOMEM */
-        return;
+    if (DEBUG_IS_SET(loglevel)) {
+        sss_vdebug_fn(__FILE__, __LINE__, "ldb", loglevel, fmt, ap);
     }
-
-    if (DEBUG_IS_SET(loglevel))
-        sss_debug_fn(__FILE__, __LINE__, "ldb", loglevel, "%s\n", message);
-
-    free(message);
 }
 
 /* In cases SSSD used to run as the root user, but runs as the SSSD user now,
diff --git a/src/util/sss_semanage.c b/src/util/sss_semanage.c
index 4fb9df589bbfddcc815ed321b6e3b32655d44a0c..7f746491174730acbf1539acaf3f6467ff92afcf 100644
--- a/src/util/sss_semanage.c
+++ b/src/util/sss_semanage.c
@@ -39,8 +39,6 @@ static void sss_semanage_error_callback(void *varg,
                                         const char *fmt, ...)
 {
     int level = SSSDBG_INVALID;
-    int ret;
-    char * message = NULL;
     va_list ap;
 
     switch (semanage_msg_get_level(handle)) {
@@ -56,16 +54,10 @@ static void sss_semanage_error_callback(void *varg,
     }
 
     va_start(ap, fmt);
-    ret = vasprintf(&message, fmt, ap);
+    if (DEBUG_IS_SET(level)) {
+        sss_vdebug_fn(__FILE__, __LINE__, "libsemanage", level, fmt, ap);
+    }
     va_end(ap);
-    if (ret < 0) {
-        /* ENOMEM */
-        return;
-    }
-
-    if (DEBUG_IS_SET(level))
-        sss_debug_fn(__FILE__, __LINE__, "libsemanage", level, "%s\n", message);
-    free(message);
 }
 
 static void sss_semanage_close(semanage_handle_t *handle)
-- 
2.5.0

